### ISSUE TITLE
[ui5-dynamicpage]: header collapse/expand and pin/unpin not announced by screenreader

### ISSUE THREAD

**Issue by ee92** [SAP/ui5-webcomponents-react#5666]
### Bug Description

The button label is read but the state change is not announced to screenreader users when they expand/collapse or pin/unpin the page header

### Affected Component

ui5-dynamicpage

### Expected Behaviour

Expect the state change to be announced

### Isolated Example

https://sap.github.io/ui5-webcomponents-react/?path=/docs/layouts-floorplans-dynamicpage--docs

### Steps to Reproduce

1. Using a screenreader (JAWS, NVDA, VoiceOver) toggle the the expand and pin buttons and notice the button label is read out but the state change is not announced

### Priority

None

### UI5 Web Components Version

1.23.0

### Browser

Chrome

### Organization

SuccessFactors

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by ilhan007** [comment]
Hi @ee92 and colleagues, the Dynamic Page belongs to this repo, thus transferring the issue.

**Comment by Lukas742** [comment]
Hi @ee92 

thanks for reporting!

The linked PR will fix the issue with the expand button, but for the pin button the behavior is already implemented according to specs:

https://github.com/SAP/ui5-webcomponents-react/assets/9749730/81882553-b917-49ef-ab7a-724f86f11e38
